### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by manu2002g**
- Add UML diagrams and implementation details for DG
- Update PPP

### CODE REVIEW

File: docs/diagrams/FindPredicatesClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ckp <|----- tckpOr
+class "{abstract}\nNameContainsKeywordPredicateAnd" as nckpAnd
+class "{abstract}\nNameContainsKeywordPredicateOr" as nckpOr
+class "{abstract}\nPhoneContainsKeywordPredicateAnd" as pckpAnd
+class "{abstract}\nPhoneContainsKeywordPredicateOr" as pckpOr
+fckpAnd <|-- nckpAnd
+fckpOr <|-- nckpOr
+fckpAnd <|-- pckpAnd
+fckpOr <|-- pckpOr
+class "{abstract}\CcaContainsKeywordPredicateAnd" as cckpAnd
```
Comment (ongkimlai) [inline docs/diagrams/FindPredicatesClassDiagram.puml]: Missing the "n" for "\n"

**Review by ongkimlai — CHANGES_REQUESTED** [review]
Error found in FindPredicateClassDiagram. 
Perhaps layout can be better too.

File: docs/diagrams/FindPredicatesClassDiagram.puml (hunk 1 above)
Comment (manu2002g) [reply docs/diagrams/FindPredicatesClassDiagram.puml]: Oh good catch thanks!

**Review by ongkimlai — APPROVED** [review]
Looks fine

**Pull request merged**